src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.cs

```diff
@@ -115,11 +115,7 @@ private Utf8Span(ReadOnlySpan<byte> rawData)
             // Allow retrieving references to just past the end of the span (but shouldn't dereference this).
 
             Debug.Assert(index <= (uint)Length, "Caller should've performed bounds checking.");
-#if SYSTEM_PRIVATE_CORELIB
             return ref Unsafe.AddByteOffset(ref DangerousGetMutableReference(), index);
```

Comment:
There are no UIntPtr overloads in the public `System.Runtime.CompilerServices.Unsafe`. I expect that you will need to keep the cast to make it compile.

We may look at adding the UIntPtr overloads for `Unsafe.Add/Subtract` in `System.Runtime.CompilerServices.Unsafe`. Now that we have the nuint type, they make a lot more sense than before.

Reply:
Do you think I should log an "api proposal" for the UIntPtr overloads?

Reply:
Sounds good to me.

Reply:
I assume that it will be no-brainer to get approved.

Reply:
cc @GrabYourPitchforks

Reply:
I see now why my local build succeeded. I built Utf8String.Experimental using the default TFM (which is `net5.0`). This code isn't compiled during the `net5.0` build.

Reply:
Logged https://github.com/dotnet/runtime/issues/36232